test/dns_nsupdate/dns_nsupdate_config_test.rb

```diff
@@ -2,38 +2,21 @@
 require 'dns/dns'
 require 'dns_nsupdate/dns_nsupdate'
 require 'dns_nsupdate/dns_nsupdate_gss'
-require 'dns_nsupdate/dns_nsupdate_main'
-require 'dns_nsupdate/dns_nsupdate_gss_main'
 
 class DnsNsupdateConfigTest < Test::Unit::TestCase
-  def test_initialize_nsupdate_returns_no_error_with_missing_key_setting
-    Proxy::Dns::Nsupdate::Plugin.settings.stubs(:dns_key).returns(nil)
-    assert Proxy::Dns::Nsupdate::Record.new(:fqdn => 'example.com')
-  end
+  def test_nsupdate_default_settings
+    Proxy::Dns::Nsupdate::Plugin.load_test_settings({})
 
-  def test_initialize_nsupdate_returns_error_with_missing_key_file
-    Proxy::Dns::Nsupdate::Plugin.settings.stubs(:dns_key).returns('./no-such-key')
-    assert_raise RuntimeError do
-      Proxy::Dns::Nsupdate::Record.new(:fqdn => 'example.com')
-    end
+    assert_equal "localhost", Proxy::Dns::Nsupdate::Plugin.settings.dns_server
+    assert_nil Proxy::Dns::Nsupdate::Plugin.settings.dns_key
   end
 
-  def test_initialize_nsupdate_gss_succeeds
-    Proxy::Dns::NsupdateGSS::Plugin.settings.expects(:dns_tsig_keytab).returns('./key')
-    Proxy::Dns::NsupdateGSS::Plugin.settings.expects(:dns_tsig_principal).returns('a@B')
-    File.expects(:exist?).with('./key').returns(true)
-    record = Proxy::Dns::NsupdateGSS::Record.new(:fqdn => 'example.com')
-    assert record
-    assert_equal './key', record.tsig_keytab
-    assert_equal 'a@B', record.tsig_principal
```

Comment:
Could we also keep tests that create an instance of the provider to test the initialiser?

For the GSS and virsh providers we should probably be testing that they retrieve and store the right keytab/principal/network settings.  It would probably still be useful for all providers to test that if the server's not passed in that they initialise it and the TTL from settings (usually the tests pass their own, right?).